### DISCUSSION TITLE
Firmware

### DISCUSSION THREAD

**Discussion by jsikken** [DatanoiseTV/PicoDSP-Hardware#7] [Q&A]
Hi there!

I just got my pico ADK and I have a very noob question.. Do I have to compile the firmware manually or can I use the firmware.uf2 file from the releases-page??

**Comment by DatanoiseTV** [comment]
Hey,

However you prefer. If you just want to test it, use the UF2 file. If you want to write your own firmware, use the firmware template and write your own code :)

**Reply by jsikken** [reply]
Thanks for the super fast reply! :)
So when I compile the firmware I can add the .vult file? Or can I add it later on using your uf2 firmware file? I keep getting an error on my Mac when I try to compile it😅

**Comment by DatanoiseTV** [comment]
If you want to write your code in vult, you have to edit vultsrc/dsp.vult and compile the firmware.
What is the error message you are getting and are you on an Intel or ARM mac?

**Reply by jsikken** [reply]
I get this error on mac and ubuntu :(

[137/221] Transcompiling Vult DSP Code to C++.
ninja: job failed: cd /project/build && /project/tools/vultc /project/vultsrc/dsp.vult -i /project/lib/vult/examples/osc -i /project/lib/vult/examples/util -i /project/lib/vult/examples/filters -i /project/lib/vult/examples/env -i /project/lib/vult/examples/midi -i /project/lib/vult/examples/effects -i /project/lib/vult/examples/units -ccode -real fixed -samplerate 44140 -o vult
ninja: subcommand failed
/bin/sh: /project/tools/vultc: not found

I have vult and docker installed

**Reply by jsikken** [reply]
*Intel mac!

**Reply by jsikken** [reply]
I also get error messages on rpi os. At 40 procent when it's transcompiling the vult dsp code to c++

**Reply by DonnaHammond** [reply]
For information,in the past 5 days I have also been trying to build the PicoADK Template , and ALSO encounter the build error at [40%] Transcompiling Vult DSP Code to C++
../tools/vultc: 8: Syntax error: ")" unexpected
Followed by 3 Error 2s
make[2]: .........Error 2

**Comment by DatanoiseTV** [comment]
I have fixed it yesterday. Can you pull and retry please?

**Reply by DonnaHammond** [reply]
Thanks for quick reply, I did a re-install with latest Pico-sdk and Pico ADK this afternoon, and the tool chain and build now work successfully. 
Kind Regards 
              Donna.
